### ISSUE TITLE
Add 'ProjectVersion' class to 'project' module

### ISSUE THREAD

**Issue by ChrisTimperley** [squaresLab/blameandshame#54]
We probably want to introduce a notion of `ProjectVersion` into blameandshame. A project version represents the state of a particular project at a given commit. Right now, we're passing that information around as a `Tuple[Project, git.Commit]`, but we should refactor those to be `ProjectVersion` instead.

**Comment by jlacomis** [comment]
This seems like it is related pretty closely to #27

**Comment by ChrisTimperley** [comment]
Yep, I guess it is. Under the hood, `ProjectVersion` should interface with BugZoo in order to run tests and to collect coverage.
